docs/core/testing/unit-testing-best-practices.md

```diff
@@ -0,0 +1,515 @@
+- Less chance of setting up too much or too little for the given test.
+- Less chance of sharing state between tests which creates unwanted dependencies between them.
+
+#### Bad:
+```csharp
+Glossary glossary;
+
+[SetUp]
+public void Initialize()
+{
+    glossary = new Glossary();
+}
+
+// more tests..
+
+public void ParseWord_NullValue_ThrowsArgumentException()
+{
+    Assert.Throws<ArgumentException>(() => glossary.ParseWord(null));
+}
+
+public void ParseWord_EmptyString_ReturnsEmptyString()
+{
+    Assert.Empty(glossary.ParseWord(""));
+}
+```
+
+#### Better:
+```csharp
+public void ParseWord_NullValue_ThrowsArgumentException()
+{
+    var glossary = CreateDefaultGlossary();
+
+    var result = () => glossary.ParseWord(null);
+
+    Assert.Throws<ArgumentException>(result);
+}
+
+public void ParseWord_EmptyString_ReturnsEmptyString()
+{
+    var glossary = CreateDefaultGlossary();
+
+    var result = glossary.ParseWord("");
+
+    Assert.Empty(result);
+}
+
+// more tests..
+
+private Glossary CreateDefaultGlossary()
+{
+    return new Glossary();
+}
+```
+
+### Avoid Multiple Asserts
+When writing your tests, try to only include one Assert per test. Common approaches to using only one assert include:
+- Create a separate test for each assert.
+- Use parameterized tests.
+
+#### Why?
+- If one Assert fails, the subsequent Asserts will not be evaluated.
+- Ensures you are not asserting multiple cases in your tests.
+- Gives you the entire picture as to why your tests are failing. 
+
+#### Bad:
+```csharp
+public void IsValidWord_InputIsNullOrEmpty_ReturnsFalse()
+{
+    var glossary = new Glossary();
+
+    Assert.False(glossary.IsValidWord(null)); // potential NullException
+    Assert.False(glossary.IsValidWord("")); // this never gets executed if the above test fails
+}
+```
+
+#### Better:
+```csharp
+[InlineData(null)]
+[InlineData("")]
+public void IsValidWord_InputIsNullOrEmpty_ReturnsFalse(string input)
+{
+    var glossary = new Glossary();
+    
+    var result = glossary.IsValidWord(input);
+
+    Assert.False(result);
+}
+```
+
+### Write Minimally Passing Tests
+The input to be used in a unit test should be the simplest possible in order to pass the behavior that you are currently testing.
+
+#### Why?
+- Tests become more resilient to future changes in the codebase.
+- Closer to testing behavior over implementation.
+
+#### Bad:
+```csharp
+public void ConcatenateWords_ByDefault_ReturnsStringWithCommaBetween()
+{
+    var glossary = new Glossary();
+    var firstWord = "aardvark";
+    var secondWord = "baboon";
+
+    var result = glossary.ConcatenateWords(firstWord, secondWord)
+
+    Assert.Equals("aardvark,baboon", result)
+}
+```
+
+#### Better:
+```csharp
+public void ConcatenateWords_ByDefault_ReturnsStringWithCommaBetween()
+{
+    var glossary = new Glossary();
+
+    var result = glossary.ConcatenateWords("a", "b")
+
+    Assert.Equals("a,b", result)
+}
+```
+
+## How Do I...?
+
+### Test Private Methods
```

Comment:
I'd change this header to move of a positive guideline.  Perhaps "Validate private methods by unit testing public methods".